src/logicsig.js

```diff
@@ -200,11 +197,24 @@ function signLogicSigTransaction(txn, lsig) {
  * @returns {Object} Object containing txID and blob representing signed transaction.
  */
 function signLogicSigTransactionObject(txn, lsig) {
-    let lstx = {
+    const lstx = {
         lsig: lsig.get_obj_for_encoding(),
         txn: txn.get_obj_for_encoding()
     };
 
+    const isDelegated = lsig.sig || lsig.msig;
+    if (isDelegated) {
+        if (!lsig.verify(txn.from.publicKey)) {
+            throw new Error("invalid signature");
```

Comment:
Consider making the error a bit more descriptive

Reply:
I will update with an improved message